pkg/v1/tkr/controllers/source/tkr_source_controller.go

```diff
@@ -347,9 +347,7 @@ func (r *reconciler) UpdateTKRCompatibleCondition(ctx context.Context, tkrs []ru
 
 	compatibleReleases := []string{}
 	for _, mgmtVersion := range metadata.ManagementClusterVersions {
-		// Fix before TKG v1.10: what if mgmtClusterVersion is "v1.10" and mgmtVersion.TKGVersion is "v1.1"?
-		// See https://github.com/vmware-tanzu/tanzu-framework/issues/452
-		if strings.HasPrefix(mgmtClusterVersion, mgmtVersion.TKGVersion) {
+		if mgmtClusterVersion == mgmtVersion.TKGVersion {
```

Comment:
Will this work for `dev`, `RC` builds as well?

Reply:
Yes, release team publishes  tkr-compatibility images with the right management cluster versions. for example for v1.5.0-rc.1 builds a seperate tkr-compatibility image would be published having the below version information

```
- version: v1.5.0-rc.1
  supportedKubernetesVersions:
  - v1.20.12+vmware.1-tkg.1-rc.1
  - v1.21.6+vmware.1-tkg.1-rc.1
  - v1.22.3+vmware.1-tkg.1-rc.1
```